math/metis/files/patch-powerpc.diff

```diff
@@ -0,0 +1,28 @@
+From 1090f554654d563ce1477f52117fa3deb6ea011f Mon Sep 17 00:00:00 2001
+From: Sergey Fedorov <vital.had@gmail.com>
+Date: Fri, 20 Jan 2023 03:28:23 +0800
+Subject: [PATCH] Do not use unsupported -march flag on PPC
+
+Signed-off-by: Sergey Fedorov <vital.had@gmail.com>
+---
+ conf/gkbuild.cmake | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/conf/gkbuild.cmake b/conf/gkbuild.cmake
+index 0e70a8e..591ece4 100644
+--- conf/gkbuild.cmake
++++ conf/gkbuild.cmake
+@@ -31,8 +31,13 @@ if(CMAKE_COMPILER_IS_GNUCC)
+   set(GK_COPTIONS "${GK_COPTIONS} -std=c99 -fno-strict-aliasing")
+ if(VALGRIND)
+   set(GK_COPTIONS "${GK_COPTIONS} -march=x86-64 -mtune=generic")
++else()
++# -march=native is not a valid flag on PPC:
++if(CMAKE_SYSTEM_PROCESSOR MATCHES "power|ppc|powerpc|ppc64|powerpc64" OR (APPLE AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc|ppc64"))
++  set(GK_COPTIONS "${GK_COPTIONS} -mtune=native")
+ else()
+   set(GK_COPTIONS "${GK_COPTIONS} -march=native")
++endif()
```

Comment:
`-march=native` should not be used on any system as a the buildbot could support different options than the machine from an end-user. So this should be switched off by default and if you want if could be added as a "variant" as is done in other ports.

Reply:
@reneeotten Sure, we can remove it. Replace by `-mtune` or remove altogether?

Reply:
I think in other ports the whole thing is removed, but I haven't checked - you could do a quick grep..

Reply:
Certainly not systemically, as I’m getting the same errors every once in a while. I do not know whether gains in performance are anything noticeable, but it may make a better sense add these variants on level of PG and not individual ports. And remove from ports.

Reply:
what PG - this is specific to certain ports so it should likely be resolved in these ports.

Reply:
Flags are supported by compilers. Anything compilation-intensive etc may be worth building in cpu-optimized way. Make a PG for that? optimizations 1.0 :)

Reply:
> Flags are supported by compilers. Anything compilation-intensive etc may be worth building in cpu-optimized way. Make a PG for that? optimizations 1.0 :)

sounds like an overkill - there aren't all that many ports that add this stuff so I doubt it will fly to make a PG for that stuff